### PR TITLE
Add all HTTP methods to route when the method is empty

### PR DESCRIPTION
**Pull request by minwoox**
Motivation:
Check out this code:
```java
sb.decorator((delegate, ctx, req) -> {
    System.err.println(2);
    return delegate.serve(ctx, req);
});
sb.routeDecorator().pathPrefix("/").build((delegate, ctx, req) -> {
    System.err.println(1);
    return delegate.serve(ctx, req);
});
sb.decoratorUnder("/", (delegate, ctx, req) -> {
    System.err.println(3);
    return delegate.serve(ctx, req);
});
```
The executed order of decorators are a little messed becuase `routeDecorator()` has all `HttpMethod`s set, but others don't.
If we set `HttpMethods.knownMethods()` when there's no method set for a route, the execution order of decorator will be the order they inserted.

Modifications:
- Add `HttpMethod.knownMethods()` to the `Route` if it's empty.
  - However, the logger name and metric don't include it because it's verbose.

Result:
- Right decoration order.

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/server/DefaultRoute.java [hunk 1]
```diff
@@ -244,7 +244,9 @@ private static String generateLoggerName(String prefix, Set<HttpMethod> methods,
                                              List<RoutingPredicate<HttpHeaders>> headerPredicates) {
         final StringJoiner name = new StringJoiner(".");
         name.add(prefix);
-        if (!methods.isEmpty()) {
+
+        // Skip if the methods is empty or it's knownMethods because it's verbose.
+        if (!methods.isEmpty() && methods != HttpMethod.knownMethods()) {
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/server/DefaultRoute.java]: (Maybe) don't we need to update `methods.isEmpty()` condition in `apply(RoutingContext routingCtx)`?
https://github.com/line/armeria/blob/3bc7bbd9363fef1a3b6d42569222ee5a4d69679d/core/src/main/java/com/linecorp/armeria/server/DefaultRoute.java#L99

Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/server/DefaultRoute.java]: Indeed! Thanks.

File: core/src/main/java/com/linecorp/armeria/server/DefaultRoute.java [hunk 2]
```diff
@@ -58,6 +59,7 @@
                  List<RoutingPredicate<QueryParams>> paramPredicates,
                  List<RoutingPredicate<HttpHeaders>> headerPredicates) {
         this.pathMapping = requireNonNull(pathMapping, "pathMapping");
+        checkArgument(!methods.isEmpty(), "methods is empty.");
         this.methods = Sets.immutableEnumSet(requireNonNull(methods, "methods"));
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/server/DefaultRoute.java]: You have to check null first.

Comment (minwoox) [reply core/src/main/java/com/linecorp/armeria/server/DefaultRoute.java]: Oops 😅

**Pull request merged**